### ISSUE TITLE
[Feature Request] store notes/extra information in database

### ISSUE THREAD

**Issue by NimdaBE** [bbilly1/tubearchivist#144]
Hi,

thank you for the great piece of software. I'm testing it out now and i have a question:

Is it possible to add a textfield / box to store notes?

Motivation: i intend to use this program to store a lot of cooking and baking videos. A lot of these videos don't put the recipe in the description box but put a link to their own website. If Tubearchivist would have a extra comments field/box where we can manually input notes then we can store the information alongside the video. That way we can be 100% sure that when a channel (together with the website) goes offline the extra information is retained...

thanks again

**Comment by bbilly1** [comment]
Yes, I can see this being useful for that. Makes sense for your example, just wondering if there are other usecases for that feature? Like, when you have a generic *Notes* section per video, what else could you use it for?

**Comment by NimdaBE** [comment]
True, my usecase is specific and i cannot seem to immediately think of other uses for such a field... One thing i was thinking of was tagging the videos with one or more labels for easier filtering and searching (creating a playlist is sometimes too limiting)

So just thinking out loud: using this comment box not only for notes, but also for hashtags to label/tag videos? And some sort of filtering system based on those tags?

**Comment by helmutundarnold** [comment]
For me labeling or tagging would be a great help - because I've start using it for our family in this regard I could label ..... mom, dad, son and lot more to find.

**Comment by 2ndLtAmazing** [comment]
Hello, (I think this is literally the first time I've ever commented on anything in GitHub.)

I really love this project and would have specific interest in this feature!
To me, the uses are endless.  Being able to add notes about a particular video about specific content matter would be invaluable.  Often times I find myself looking through 50 videos to see which one has the specific reference that I want to show somebody.  Having the ability to add notes would be amazingly helpful for this.

Suppose I'm looking through videos of a particular person that may have a hundred videos on a similar topic, but only one time do they specifically say "And that's why you should never actually...." but all I'm stuck with is knowing that they said it, and it was in a video I watched at some point.

Now already being able to search through transcripts does make this easier by itself, but sometimes they are not accurate, and many videos don't have them at all.  Sometimes the thing I want to keep track of is more abstract.  Sometimes it isn't something that's spoken.  Also, sometimes I may have multiple similar videos that I might want to preserve the comments from, but only 1 of the 3 actually has the specific part in it that I would want to reference later, so I could notate something even as simple as "this is the one" rather than remembering somehow which was which without actually scrubbing through each video manually every time.

I might for example also save multiple news sources that cover the same story, and I could make notes about specific details that were included or left out.

Here's a big one, "This video is factually incorrect at 22:30 he says [whatever] but he actually made another video [insert link to that video here in the notes for this video] where he explained what that was inaccurate.] but the rest of the content of the video is certainly still worth keeping.   Maybe I might look up a video like that for a reference on how to do something later, but I forgot that one of the steps actually causes problems, and so I could make a note about it at the time.

Also, maybe there is a video that is an episode of a podcast has 2 people talking and at one particular point one of the people says something absurd, and the other one gives him an hilarious look.  I could put the note saying "funny look at 1:17:19" because 5 months from now when I want to show somebody, I'll forget which episode it was on, let alone what the timestamp was which basically makes it impossible to find if I have 40 episodes that are each 2 hours long.

It would also be useful to make basic study notes if it's a topic that I'm trying to learn about.

Perhaps it's a video that expresses a political or philosophical idea that I want to cross reference with another video and I could make notes about which ones are related.  Playlists could be used for this but it seems like a really complicated way to say "these 2 videos, and only these 2 videos touch on a particular topic that I've never seen anybody else talk about before" and then I'd have to come up with a meaningful title for the playlist, and remember that I created the playlist in the first the next time I'm trying to find the video or the next time I'm watching the video if I want to be reminded of whatever it was. 
(Note that "remembering" being a prerequisite of "being reminded" is a self-defeating plan, and a pretty good example of the whole idea of what being able to add notes could do.)

Again, in something like a news broadcast, searching through the transcript, which may not be transcribed correctly in the first place, still doesn't make it easy to find "that one part of that one video" instantly months later when I'm trying to remember it like "I remember that they were talking about something... and later referenced it... and it was hilarious.... and I don't quite remember what they even said... "
And here I might already have the video in a "Funny News Things" playlist, but there's 900 other videos in there, so that doesn't help narrow it down either.


TLDR:  There are endless uses for this feature and I intend to become a contributor to this project in the hopes of accelerating its implementation (even if it must come after other more urgent updates).

**Comment by PawsFunctions** [comment]
> Suppose I'm looking through videos of a particular person that may have a hundred videos on a similar topic, but only one time do they specifically say "And that's why you should never actually...." but all I'm stuck with is knowing that they said it, and it was in a video I watched at some point.
> 
Being able to add notes would be a really good feature. Right now, I search through subtitles.
